src/schemathesis/models.py

```diff
@@ -176,10 +177,13 @@ def call(
             close_session = True
         else:
             close_session = False
-
         data = self.as_requests_kwargs(base_url)
+        data["headers"] = data["headers"] or {}
+        data["headers"] = data["headers"].copy()
         if headers is not None:
-            data["headers"] = {**(data["headers"] or {}), **headers}
+            data["headers"].update(headers)
+        if "user-agent" not in set(k.lower() for k in data["headers"]):
```

Comment:
I'd prefer to keep the current behavior for consistency and limiting the scope of this PR

```suggestion
        if "User-Agent" not in data["headers"]:
```